demos/mcp_server_memory/db/sqlite_manager.py

```
@@ -0,0 +1,581 @@
+        except Exception as e:
+            print(f"Error creating directory: {e}")
+
+    def _query_fetch(self, query: str, all: bool = True) -> list | None:
+        try:
+            with SQLiteConnection(SQLITE_PATH) as conn:
+                cursor = conn.cursor()
+                cursor.execute(query)
+                if all:
+                    result = cursor.fetchall()
+                else:
+                    result = cursor.fetchone()
```

Ruff (0.11.9)

Use ternary operator `result = cursor.fetchall() if all else cursor.fetchone()` instead of `if`-`else`-block

Replace `if`-`else`-block with `result = cursor.fetchall() if all else cursor.fetchone()`

(SIM108)